dao/src/main/java/org/thingsboard/server/dao/notification/DefaultNotificationSettingsService.java

```diff
@@ -81,6 +95,50 @@ public NotificationSettings findNotificationSettings(TenantId tenantId) {
                 });
     }
 
+    @Override
+    public UserNotificationSettings saveUserNotificationSettings(TenantId tenantId, UserId userId, UserNotificationSettings settings) {
+        User user = userService.findUserById(tenantId, userId);
+        ObjectNode additionalInfo = (ObjectNode) Optional.ofNullable(user.getAdditionalInfo()).orElseGet(JacksonUtil::newObjectNode);
+        additionalInfo.set(USER_SETTINGS_KEY, JacksonUtil.valueToTree(settings));
+        user.setAdditionalInfo(additionalInfo);
+        userService.saveUser(user);
+        return formatUserNotificationSettings(settings);
+    }
```

Comment:
Settings are stored inside additionalInfo to minimize the amount of additional DB calls.

Reply:
See ```org.thingsboard.server.common.data.settings.UserSettings```. We have them cached. Please add a new ```UserSettingsType``` and consider using them.